distributed/deploy/ssh.py

```diff
@@ -256,8 +264,9 @@ def __init__(self, scheduler_addr, scheduler_port, worker_addrs, nthreads = 0, n
 
         # Start worker nodes
         self.workers = []
-        for i, addr in enumerate(worker_addrs):
-            self.add_worker(addr)
+        for addr, nthreads, nprocs in zip(worker_addrs, self.threads,
+                                          self.nprocs):
+            self.add_worker(addr, nthreads, nprocs)
```

Comment:
Ideally we would make this general to all keywords.  I expect users to become confused why only certain keywords are respected in this way.